pypit/arproc.py

```diff
@@ -1721,6 +1752,50 @@ def lacosmic(slf, fitsdict, det, sciframe, scidx, maxiter=1, grow=1.5, maskval=-
     return crmask
 
 
+def new_cr_screen(a, mask_value=0.0, spatial_axis=1):
+    r"""
+    Calculate the significance of pixel deviations from the median along
+    the spatial direction.
+
+    No type checking is performed of the input array; however, the
+    function assumes floating point values.
+
+    Args:
+        a (numpy.ndarray): Input 2D array
+        mask_value (float): (**Optional**) Values to ignore during the
+            calculation of the median.  Default is 0.0.
+        spatial_axis (int): (**Optional**) Axis along which to calculate
+            the median.  Default is 1.
+
+    Returns:
+        numpy.ndarray: Returns a map of :math:`|\Delta_{i,j}|/\sigma_j`,
+        where :math:`\Delta_{i,j}` is the difference between the pixel
+        value and the median along axis :math:`i` and :math:`\sigma_j`
+        is robustly determined using the median absolute deviation,
+        :math:`sigma_j = 1.4826 MAD`.
+
+    Raises:
+        ValueError: Raised if the input array is not two dimensional or
+            if the selected axis is neither 0 nor 1.
+    """
+    # Check input
+    if len(a.shape) != 2:
+        raise ValueError('Input array must be two-dimensional.')
```

Comment:
In three cases of this PR, a ValueError is raised. Should we instead take advantage of the msgs.error() function?

Reply:
Not a problem.  I'm not yet up to speed on pypit's exception handling code, but this is an easy change.